File: 28_feb/maxMinAvg/maxMinAvg.py
def get_max_min_avg(arr):
	n = len(arr)
	max_min_avg = {
		"max": arr[0],
		"min": arr[0],
		"avg": arr[0]
	}

	for i in range(n):
		if arr[i] > max_min_avg["max"]:
			max_min_avg["max"] = arr[i]
		if arr[i] < max_min_avg["min"]:
			max_min_avg["min"] = arr[i]
		max_min_avg["avg"] += arr[i]

	max_min_avg["avg"] /= float(n)

	return max_min_avg

inp_arr = [5, 8, 2, 14, 25, 9]

print get_max_min_avg(inp_arr)

File: 28_feb/maxMinAvg/maxMinAvg.cpp
#include <map>
#include <string>
#include <utility>
#include <iostream>
#include <vector>

std::map<std::string, float> getMaxMinAvg(std::vector<int> arr);

std::map<std::string, float> maxMinAvg;
std::vector<int> inpArr;

int main(int argc, char const *argv[])
{
	inpArr.push_back(5);
	inpArr.push_back(8);
	inpArr.push_back(2);
	inpArr.push_back(14);
	inpArr.push_back(25);
	inpArr.push_back(9);

	getMaxMinAvg(inpArr);

	return 0;
}

std::map<std::string, float> getMaxMinAvg(std::vector<int> arr)
{
	maxMinAvg.insert(std::make_pair("Max", (float) arr[0]));
	maxMinAvg.insert(std::make_pair("Min", (float) arr[0]));
	maxMinAvg.insert(std::make_pair("Avg", (float) arr[0]));

	int n = arr.size(), i;

	for (i = 1; i < n; i++)
	{
		if (arr[i] > maxMinAvg["Max"])
		{
			maxMinAvg["Max"] = arr[i];
		}
		if (arr[i] < maxMinAvg["Min"])
		{
			maxMinAvg["Min"] = arr[i];
		}
		maxMinAvg["Avg"] += arr[i];
	}

	maxMinAvg["Avg"] /= (float)n;



	for (std::map<std::string, float>::iterator it = maxMinAvg.begin(); it != maxMinAvg.end(); it++)
	{
		std::cout << it -> first << ": " << it -> second << std::endl;
	}

	return maxMinAvg;
}
